packages/amplify-cli/src/input-manager.ts

```diff
@@ -63,19 +64,47 @@ export function getCommandLineInput(pluginPlatform: PluginPlatform): Input {
   return result;
 }
 
+function preserveHelpInformation(input: Input): Input {
```

Comment:
I'm having a hard time following this function. Can it be refactored so that its purpose is clearer?